api/src/services/xml_generation/value_transformers.py

```diff
@@ -0,0 +1,175 @@
+
+    Returns:
+        Formatted currency string
+
+    Raises:
+        ValueTransformationError: If currency formatting fails
+    """
+    if isinstance(value, (int, float)):
+        # Convert number to string with proper decimal places
+        if ensure_decimal:
+            return f"{value:.2f}"
+        else:
+            return str(value)
+
+    elif isinstance(value, str):
+        # Validate and potentially reformat string
+        # Remove any non-numeric characters except decimal point
+        cleaned = re.sub(r"[^\d.]", "", value)
+
+        if not cleaned:
+            raise ValueTransformationError(f"Invalid currency value: '{value}'")
+
+        try:
+            # Convert to float and back to ensure proper format
+            float_value = float(cleaned)
+            if ensure_decimal:
+                return f"{float_value:.2f}"
+            else:
+                return str(float_value)
+        except ValueError as e:
+            raise ValueTransformationError(f"Cannot parse currency value: '{value}'") from e
+
+    else:
+        raise ValueTransformationError(
+            f"Currency value must be string or number, got {type(value)}"
+        )
+
+
+def transform_string_case(value: Any, case_type: str) -> str:
+    """Transform string case.
+
+    Args:
+        value: String value to transform
+        case_type: Type of case transformation ("upper", "lower", "title")
+
+    Returns:
+        Transformed string
+
+    Raises:
+        ValueTransformationError: If transformation fails
+    """
+    if not isinstance(value, str):
+        raise ValueTransformationError(
+            f"String transformation requires string input, got {type(value)}"
+        )
+
+    if case_type == "upper":
+        return value.upper()
+    elif case_type == "lower":
+        return value.lower()
+    elif case_type == "title":
+        return value.title()
+    else:
+        raise ValueTransformationError(f"Unsupported case type: {case_type}")
+
+
+def transform_truncate_string(value: Any, max_length: int) -> str:
+    """Truncate string to maximum length.
+
+    Args:
+        value: String value to truncate
+        max_length: Maximum allowed length
+
+    Returns:
+        Truncated string
+
+    Raises:
+        ValueTransformationError: If transformation fails
+    """
+    if not isinstance(value, str):
+        raise ValueTransformationError(
+            f"String truncation requires string input, got {type(value)}"
+        )
+
+    if max_length < 0:
+        raise ValueTransformationError(f"Max length must be non-negative, got {max_length}")
+
+    return value[:max_length] if len(value) > max_length else value
+
+
+# Registry of available transformations
+TRANSFORMATION_REGISTRY: dict[str, Callable[..., Any]] = {
+    "boolean_to_yes_no": transform_boolean_to_yes_no,
+    "currency_format": transform_currency_format,
+    "string_case": transform_string_case,
+    "truncate_string": transform_truncate_string,
+}
```

Comment:
How do these transforms handle `None`? Is that handled before the transform functions get called?

Guessing we'll want a way to configure how `None` behaves in the config for a given field, I could see one of two ways we'd want it to work:
* Exclude the field entirely from the XML output
* Include the field, set to null

I think the first of those would be the most common / could be the default?